src/main/java/seedu/eduke8/question/TopicQuestionList.java

```diff
@@ -0,0 +1,53 @@
+package seedu.eduke8.question;
+
+import seedu.eduke8.common.Displayable;
+import seedu.eduke8.common.DisplayableList;
+import seedu.eduke8.exception.Eduke8Exception;
+
+import java.util.ArrayList;
+
+public class TopicQuestionList implements DisplayableList {
+
+    private ArrayList<Displayable> questions;  // list of questions for the particular topic
+
+    public TopicQuestionList(ArrayList<Displayable> questions) {
+        this.questions = questions;
+    }
+
+    public QuizQuestionList getQuizQuestionList(int numberOfQuestionsForQuiz) throws Eduke8Exception {
```

Comment:
I think you should try not to depend on this class in to create the QuizQuestionList object as this violates SRP

Reply:
Working on it!